### PR TITLE
Finished up to step 4 of Public Feed

### PR DESCRIPTION
**Pull request by aokennard**

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+
+/**
+ * Handles fetching all messages for the public feed.
+ */
+@WebServlet("/feed")
+public class MessageFeedServlet extends HttpServlet{
+  
+
+ private Datastore datastore;
```
Comment (dgalbraith33) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: Let's indent with 2 spaces to keep things consistent across files.

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -45,27 +45,30 @@ public void storeMessage(Message message) {
 
     datastore.put(messageEntity);
   }
-
+
```
[the remark this answers is not in the document]
Comment (aokennard) [reply src/main/java/com/google/codeu/data/Datastore.java]: Resolved

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -45,27 +45,30 @@ public void storeMessage(Message message) {
...
+   * Gets messages based on a particular query and user (optional).
+   * 
+   * @return a list of messages based on the input query and user, empty list if there are no messages
+   * posted matching the query and user. If user is blank, then get messages from all users matching the query.
+   *  Output list is sorted by time, descending order.
    */
-  public List<Message> getMessages(String user) {
+  public List<Message> getMessagesByQuery(Query query, String user) {
     List<Message> messages = new ArrayList<>();
+    boolean useAnyMessage = false;
```
[the remark this answers is not in the document]
Comment (aokennard) [reply src/main/java/com/google/codeu/data/Datastore.java]: Resolved

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.codeu.data.Datastore;
+import com.google.codeu.data.Message;
+import com.google.gson.Gson;
+
+
```
[the remark this answers is not in the document]
Comment (aokennard) [reply src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: Removed double spaces here and elsewhere

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java (hunk 1 above)
Comment (aokennard) [reply src/main/java/com/google/codeu/servlets/MessageFeedServlet.java]: Resolved

File: src/main/webapp/js/feed.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+
+  // Fetch messages and add them to the page.
```
Comment (dgalbraith33) [inline src/main/webapp/js/feed.js]: It looks like there are 2-3 spaces in front of every line in this file. Let's remove the extras and keep the indentation consistent.

**Pull request merged**